app/code/Magento/Backend/Controller/Adminhtml/Auth/Login.php

```diff
@@ -69,4 +97,26 @@ private function getRedirect($path)
         $resultRedirect->setPath($path);
         return $resultRedirect;
     }
+
+    /**
+     * Verify if correct backend uri requested.
+     *
+     * @return bool
+     */
+    private function isValidBackendUri(): bool
+    {
+        $requestUri = $this->getRequest()->getRequestUri();
+        $backendApp = $this->backendAppList->getCurrentApp();
+        $baseUrl = parse_url($this->backendUrlFactory->create()->getBaseUrl(), PHP_URL_PATH);
+        if (!$backendApp) {
+            $backendFrontName = $this->frontNameResolver->getFrontName();
+        } else {
+            //In case of application authenticating through the admin login, the script name should be removed
+            //from the path, because application has own script.
+            $baseUrl = Http::getUrlNoScript($baseUrl);
```

Comment:
can we avoid using static method call here?

Reply:
@ihor-sviziev I think only by copying this method code.

Reply:
Could you explain why can't we create an object via factory and run the method from it?